File: JAVA/557. Reverse Words in a String III.java
import java.util.*;

class Solution {
    public String reverseWords(String s) {
        List<String> a = new ArrayList<>();
        String b = "";
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == ' ') {
                a.add(b);  
                b = "";    
            } else {
                b += s.charAt(i); 
            }
        }
        a.add(b); 
        b = ""; 
        for (int i = 0; i < a.size() - 1; i++) {
            b += new StringBuilder(a.get(i)).reverse().toString(); 
            b += " "; 
        }
        b += new StringBuilder(a.get(a.size() - 1)).reverse().toString();
        return b;
    }
}

File: C++/557. Reverse Words in a String III.cpp
class Solution {
public:
    string reverseWords(string s) {
        vector<string> a;
        string b="";
        for(char c:s){
            if(c==' '){
            a.push_back(b);
            b="";
            }
            else
            b+=c;
        }
        a.push_back(b);
        b="";
        for(int i=0;i<a.size()-1;i++){
            reverse(a[i].begin(),a[i].end());
            b=b+a[i]+" ";
        }
        reverse(a[a.size()-1].begin(),a[a.size()-1].end());
        b+=a[a.size()-1];
        return b;
    }
};